### ISSUE TITLE
compilation warnings: ISO C90 forbids mixing declarations and code

### ISSUE THREAD

**Issue by alzix** [libuv/libuv#3131]
* **Version**: v1.41.0
* **Platform**: `Darwin 19.6.0 Darwin Kernel Version 19.6.0: Tue Jan 12 22:13:05 PST 2021; root:xnu-6153.141.16~1/RELEASE_X86_64 x86_64`

compilation warning emitted when building with following flags enabled: `-Wall -Wextra -pedantic`
these warning prevent me from setting `-Werror`

1. 
```
libuv/src/unix/udp.c:929:3: warning: ISO C90 forbids mixing declarations and code [-Wdeclaration-after-statement]
  STATIC_ASSERT(sizeof(mreq.gsr_group) >= sizeof(*multicast_addr));
  ^
In file included from libuv/src/unix/udp.c:23:
In file included from libuv/src/unix/internal.h:25:
libuv/src/uv-common.h:61:8: note: expanded from macro 'STATIC_ASSERT'
  void uv__static_assert(int static_assert_failed[1 - 2 * !(expr)])
       ^
1 warning generated.
```

2.
```
libuv/src/unix/darwin.c:260:26: warning: ISO C90 forbids mixing declarations and code [-Wdeclaration-after-statement]
  CFMutableDictionaryRef classes_to_match
                         ^
1 warning generated.
```

**Comment by cjihrig** [comment]
Thanks for reporting this. Care to send a PR?

**Comment by cabelitos** [comment]
Hello @alzix and @cjihrig, I was compiling libuv on darwin and I found this problem as well.
I've created a pull request which fixes this problem and also fixes #3161.

I did not want to get into your way @alzix , but if you want I can drop the code and send yours.

**Comment by alzix** [comment]
@cabelitos thank you for the offer - i'll wait an official release
